### PR TITLE
[WFGP-34] Provisioning hangs when a batch operation fails

### PR DESCRIPTION
**Pull request by yersan**
An Exception thrown from the main method of the forked process is not correctly read by the caller which hangs waiting for a response.

Jira issue: https://issues.jboss.org/browse/WFGP-34

### CODE REVIEW

File: galleon-plugins/src/main/java/org/wildfly/galleon/plugin/server/ForkedEmbeddedUtil.java [hunk 1]
```diff
@@ -96,7 +98,13 @@ public static int fork(ForkCallback callback, Path props, String... args) throws
                     e.printStackTrace();
                 }
             }
-            return p.exitValue();
+
+            int exitCode = p.exitValue();
+            if (exitCode == FAILED){
```
Comment (aloubyansky) [inline galleon-plugins/src/main/java/org/wildfly/galleon/plugin/server/ForkedEmbeddedUtil.java]: Let's use if (exitCode != 0) instead. And let's modify the method to not return an int but void. This exit code outside this method doesn't seem to be useful for anything.

File: galleon-plugins/src/main/java/org/wildfly/galleon/plugin/server/ForkedEmbeddedUtil.java [hunk 2]
```diff
@@ -117,32 +125,38 @@ public static Path storeSystemProps() throws ProvisioningException {
...
+                o = cls.newInstance();
+            } catch (Exception e) {
+                throw new ProvisioningException("Failed to instantiate " + args[1], e);
+            }
+            ((ForkCallback)o).forkedForEmbedded(args.length == 2 ? new String[0] : Arrays.copyOfRange(args, 2, args.length));
+
+            System.exit(NORMAL);
+        } catch (Throwable t){
+            t.printStackTrace(System.err);
+            System.exit(FAILED);
```
Comment (aloubyansky) [inline galleon-plugins/src/main/java/org/wildfly/galleon/plugin/server/ForkedEmbeddedUtil.java]: Let's avoid System.exit(...) calls. Simply re-throw the throwable from the catch.

Comment (yersan) [reply galleon-plugins/src/main/java/org/wildfly/galleon/plugin/server/ForkedEmbeddedUtil.java]: @aloubyansky I would say re-throw the exception was the root of the hang. As far as I can understand, the caller doesn't receive the signal about the process has finished if we throw an exception from the main method. Basically, the caller hangs reading the input stream.

**Comment by yersan** [comment]
Closed in favor of https://github.com/wildfly/galleon-plugins/pull/34

**Pull request closed without merging**